pkg/util/pki/generate.go

```diff
@@ -122,11 +146,18 @@ func EncodePrivateKey(pk crypto.PrivateKey, keyEncoding v1.PrivateKeyEncoding) (
 			return EncodePKCS1PrivateKey(k), nil
 		case *ecdsa.PrivateKey:
 			return EncodeECPrivateKey(k)
+		case ed25519.PrivateKey:
+			return EncodePKCS8PrivateKey(k)
 		default:
 			return nil, fmt.Errorf("error encoding private key: unknown key type: %T", pk)
 		}
 	case v1.PKCS8:
-		return EncodePKCS8PrivateKey(pk)
+		switch k := pk.(type) {
+		case *ed25519.PrivateKey:
+			return EncodePKCS8PrivateKey(*k)
+		default:
+			return EncodePKCS8PrivateKey(pk)
+		}
```

Comment:
suggestion: I think with my above suggestions this should be `ed25519.PrivateKey` (i.e. not a pointer) and so the original line would work without the need to check the type and dereference the pointer.

```suggestion
		return EncodePKCS8PrivateKey(pk)
```